precompile/modules/initia_stdlib/sources/dex.move

```diff
@@ -981,14 +1002,98 @@ module initia_std::dex {
         coin::deposit(signer::address_of(account), return_coin);
     }
 
+    /// FlashSwap is a special swap that allows the user to use return coin first and repay the offer coin later.
+    /// The borrower should repay the offer coin by calling `repay_flash_swap` function with FlashSwapLoan object.
+    /// 
+    /// https://move-book.com/programmability/hot-potato-pattern.html
+    public fun flash_swap(
+        pair: Object<Config>,
+        offer_coin: Object<Metadata>,
+        offer_amount: u64,
+        min_return: Option<u64>,
+    ): (FungibleAsset, FlashSwapLoan) acquires Config, Pool, FlashSwap {
+        // zero offer amount would be invalidated in swap_simulation
+        let return_amount = get_swap_simulation(pair, offer_coin, offer_amount);
+        assert!(
+            option::is_none(&min_return)
+                || *option::borrow(&min_return) <= return_amount,
+            error::invalid_state(EMIN_RETURN)
+        );
+
+        let pair_addr = object::object_address(&pair);
+        let pool = borrow_global<Pool>(pair_addr);
+        let pool_config = borrow_global<Config>(pair_addr);
+        let pair_signer = &object::generate_signer_for_extending(&pool_config.extend_ref);
+        let pair_addr = signer::address_of(pair_signer);
+
+        // check recursive flash swap
```

Comment:
liquidswap locks a pool when a flashswap is happening https://github.com/pontem-network/liquidswap/blob/main/sources/swap/liquidity_pool.move#L389

do u think we want to force the same check? if so we probably dont need to check recursive flash swap?

Reply:
Yea FlashSwap resource is for that! I left comment in the above review comment.

Reply:
do you think there's any security issue if we prevent flashswap -> flashswap, but still allow flashswap -> swap/add_liquidity/remove_liquidity -> repay_flashswap?

believe this is the major difference between this and liquidswap impl

Reply:
yea also thought about that, and agree good to follow their approach